### PR TITLE
Fix CTAS bugs

### LINKED ISSUE #2394: CTAS Data type of first column cannot be DOUBLE
### Steps to reproduce the behavior (Required)
```
mysql> create table test(c1 varchar(10),c2 varchar(10)) DISTRIBUTED BY HASH(c1) BUCKETS 8 PROPERTIES ("replication_num" = "1" );
Query OK, 0 rows affected (0.02 sec)

mysql> insert into test values(1,2);
Query OK, 1 row affected (0.05 sec)
{'label':'insert_547ba0b9-63d3-11ec-a5da-00163e10c9d6', 'status':'VISIBLE', 'txnId':'17055'}
mysql> select c1+c2 as cr from test;
+------+
| cr   |
+------+
|    3 |
+------+
1 row in set (0.01 sec)

mysql> create table test6 as select c1+c2 as cr from test;
ERROR 1064 (HY000): Data type of first column cannot be DOUBLE
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - 2.1

### PR DESCRIPTION
**Pull request by Astralidea**
#2393 #2394
1. SELECT statement in CTAS may be a SQL without an alias. CTAS should support the SQL as well. 
2. Because double is not precise, the key column in create table is replaced using decimal.

### CODE REVIEW

**Comment by Astralidea** [comment]
run all

**Comment by Astralidea** [comment]
run all

**Comment by Astralidea** [comment]
run all

**Comment by Astralidea** [comment]
un all

**Comment by Astralidea** [comment]
run all

**Review by chaoyli — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CTASAnalyzer.java [hunk 1]
```diff
@@ -96,6 +98,8 @@ public Relation transformCTASStmt(CreateTableAsSelectStmt createTableAsSelectStm
             Type type = allFields.get(i).getType();
             if (PrimitiveType.VARCHAR == type.getPrimitiveType()) {
                 type = stringType;
+            } else if (PrimitiveType.DOUBLE == type.getPrimitiveType()) {
+                type = Type.DEFAULT_DECIMAL128;
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CTASAnalyzer.java:102]: It's better to add some unit tests for these cases.

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/CTASAnalyzerTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                .withTable("create table test(c1 varchar(10),c2 varchar(10)) DISTRIBUTED BY HASH(c1) " +
+                        "BUCKETS 8 PROPERTIES (\"replication_num\" = \"1\" );")
+                .withTable("create table test3(c1 varchar(10),c2 varchar(10)) DISTRIBUTED BY HASH(c1) " +
+                        "BUCKETS 8 PROPERTIES (\"replication_num\" = \"1\" );");
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        File file = new File(runningDir);
+        file.delete();
```
Comment (wyb) [inline fe/fe-core/src/test/java/com/starrocks/sql/analyzer/CTASAnalyzerTest.java:46]: If there are files in the directory, delete will fail.

**Review by wyb — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Pull request merged**